### DISCUSSION TITLE
Deck of cards

### DISCUSSION THREAD

**Discussion by TomJGooding** [Textualize/textual#3072] [Show and tell]
I'm still not sure this toy project is really worthy of sharing here, but I've been working on creating a [deck of cards in the terminal](https://github.com/TomJGooding/deck-of-cards).

This has been a fun challenge to implement in Textual, including [dragging widgets](https://github.com/Textualize/textual/discussions/2752) with the mouse, [stacking widgets](https://github.com/Textualize/textual/discussions/3027) using layers, and [animating offsets](https://github.com/Textualize/textual/issues/3028#issuecomment-1663930856).

[EDIT: I forgot to mention also [differentiating a mouse click from a click-and-drag](https://github.com/TomJGooding/deck-of-cards/pull/16). I was caught out by this, so linking my fix in case it helps someone else]

I'm sure some of the code could be optimised, so any feedback would be much appreciated!

This project was inspired by [usolitaire](https://github.com/eliasdorneles/usolitaire), which you should check out if you haven't already!

**Comment by willmcgugan** [comment]
NIce!

You could build any number of card games with this. :)

@davep @darrenburns

**Reply by TomJGooding** [reply]
Thanks Will!  This was really an experiment for dragging and stacking widgets in Textual, which I still can't quite believe is working in the terminal! Potentially this might have other applications beyond just card games, which is why I eventually decided to share this here.

**Reply by davep** [reply]
Oh aye this be fun!
